### PR TITLE
Remove react-addons-perf dependency

### PR DESCRIPTION
**Pull request by pgoldtho**
### Description

Removes the dependency between swagger-ui and react-addons-perf.

### Motivation and Context

Facebook discontinued support for react-addons-perf in version 15.5 (https://reactjs.org/blog/2017/04/07/react-v15.5.0.html#discontinuing-support-for-react-addons). It wasn't included in the 15.6.2 release that switched from `BSD + Patents` to `MIT` license.  

### How Has This Been Tested?

Built and tested locally using `npm run dev` also ran `npm run test` and `npm run e2e-cypress`

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [x ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [ x] are not breaking changes.

### Documentation
- [x ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by pgoldtho** [comment]
@shockey is this likely to be approved?  The `BSD + Patents` license is causing issues for us

**Comment by shockey** [comment]
@pgoldtho this is a fair point, shame that Facebook didn't ever bump it to the new license. I'm looking at ways we can retain this for development but prevent it from being included in production builds - should have something for you today!

**Comment by shockey** [comment]
Interesting! `react-addons-perf` is a whopping [one-line wrapper](https://unpkg.com/react-addons-perf@15.4.2/index.js) around `react-dom`.

I've added a commit to your PR that retains the development-only `window.Perf` assignment, but changes it to reach into `react-dom` directly for ReactPerf, so we can eliminate the unkindly-licensed module.

File: make-webpack-config.js [hunk 1]
```diff
@@ -72,7 +72,9 @@ module.exports = function(rules, options) {
       new UglifyJsPlugin({
         uglifyOptions: {
           mangle: specialOptions.mangle,
-          compress: specialOptions.mangle,
+          compress: specialOptions.mangle ? {
+            dead_code: true
```
Comment (shockey) [inline make-webpack-config.js]: This option ensures that the development-only condition is completely removed from production builds, along with the modules needed in that block:

https://github.com/swagger-api/swagger-ui/blob/8480c13b8e2f1425e048e6f1cf30e837d7c35ab8/src/core/index.js#L10-L12

**Comment by shockey** [comment]
Merged! This will be released tomorrow 👍

**Comment by pgoldtho** [comment]
Thanks @shockey.  I don't see the release on the releases page.  Am I looking in the wrong location?

**Pull request merged**